sstables/sortedmap_key_index.go

```diff
@@ -0,0 +1,127 @@
+package sstables
+
+import (
+	"bytes"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+
+	rProto "github.com/thomasjungblut/go-sstables/recordio/proto"
+	"github.com/thomasjungblut/go-sstables/skiplist"
+	"github.com/thomasjungblut/go-sstables/sstables/proto"
+	"github.com/wk8/go-ordered-map/v2"
+)
+
+// SortedMapIndex is keeping the entire index as a slice in memory and uses binary search to find the given keys.
+type SortedMapIndex struct {
+	index *orderedmap.OrderedMap[string, IndexVal]
```

Comment:
`[]byte` should be comparable as well, if you want to use that instead of hex encoding

Reply:
![image](https://github.com/user-attachments/assets/2eb404c2-4ebc-4b7f-bb98-6842a7108096)
 
unfortunately not. But the hex encode is not the good choice i Think

Reply:
TIL, the language specification mentions `Array types are strictly comparable if their array element types are strictly comparable.`. ... `Slice, map, and function types are not comparable.`

`[1]byte` is comparable, the slice type `[]byte` isn't.

Reply:
i had some problem with that on simpledb too. I had some Get and Put with []byte keys. In some case, the string transformation don't work as expected when the bytes are binary not "unicode"

Reply:
https://github.com/golang/go/issues/38377 the comparable issue for []byte . Maybe in Go 2.0 ?

Reply:
>i had some problem with that on simpledb too. I had some Get and Put with []byte keys. In some case, the string transformation don't work as expected when the bytes are binary not "unicode"

I see, yeah that's bad. I was holding on to simple keys like hash strings or paths and json as values. Thanks for the feedback, maybe we can just add the same functions with the byte slices.

Reply:
I will contribute my Bytes function in another PR.